### PR TITLE
Chore: update react/ja/simple-component.md

### PR DESCRIPTION
**Pull request by tomoyuki-hiranuma**
Hi, this is my first OSS contribution in my life.

I've updated the Japanese translation  of "Intro to Storybook" in React series.
There were some differences between Japanese version and English version in the simple-component.md.

Thank you for being reviewed it :)

### CODE REVIEW

**Comment by jonniebigodes** [comment]
@tomoyuki-hiranuma, Wow, that's great to hear, and thank you very much for choosing us for your first contribution ever. We're thrilled to have you here helping us with the translation. From my end, this looks almost good. I wanted to ask you the following: Are you interested in updating the rest of the translation, as it's now a bit outdated? As there are still some chapters that require some additional changes. Applying them would help you out by learning how to get started on contributions to open source and also be a part of an amazing community, and the Japanese readers/writers would benefit from the changes tremendously.

Let me know if you're interested in doing them, you can use this pr and push in the changes. If by any reason you don't have the time, that's ok. I'll reach out to the community and ask for someone to proofread this and merge it in.

Hope you have a great week

Stay safe

**Comment by tomoyuki-hiranuma** [comment]
@jonniebigodes , Thank you for replying and answer the questions. 

>  Are you interested in updating the rest of the translation, as it's now a bit outdated?
Of course!! I'd like to update the rest of the outdated translation and help Japanese readers/writers through my contribution.

I will push the changes to this PR sooner or later.

Thank you for asking for someone to proofread.

Stay safe

**Comment by jonniebigodes** [comment]
@tomoyuki-hiranuma, that's great to hear. Please do let me know once the changes are up so that I can then reach out to the community for a translator and get them addressed, if any, and merge the updates.

Have a great weekend.

**Comment by tomoyuki-hiranuma** [comment]
@jonniebigodes , I did update the translation in two more files that I found the outdated.
Thank you for waiting my contribution.

**Comment by jonniebigodes** [comment]
@chameleonhead sorry to bother you once again, but could you take a look at this one as well and see if it's ok, I've checked on my end (code), and it looks good. Waiting on the wording so we can merge it and get the first translation updated to feature interaction testing and A11y testing. Let me know, and we'll go from there.

@tomoyuki-hiranuma sorry for the delayed response, but it's been a couple of busy weeks, and I wasn't able to follow up with you.


Looking forward to hearing from both of you and seeing this get merged in, and seeing the first translation updated.

Hope both of you have have a great day.

Stay safe

File: content/intro-to-storybook/react/ja/test.md [hunk 1]
```diff
@@ -41,18 +41,19 @@ Storybook は視覚的なリグレッションテスト用の素晴らしいツ
 git checkout -b change-task-background
 ```
 
-`Task` を以下のように変更します:
```
Comment (chameleonhead) [inline content/intro-to-storybook/react/ja/test.md:44]: このページの上部にある「スナップショットテスト」→「アクセシビリティテスト」、ともう一つの変更も反映してもらえると良いと思います。

> - **Accessibility tests** with a11y addon verify that the component is accessible to everyone. They're great for allowing us to collect information about how people with certain types of disabilities use our components.
> - **Interaction tests** with the play function verify that the component behaves as expected when interacting with it. They're great for testing the behavior of a component when it's in use.

File: content/intro-to-storybook/react/ja/test.md [hunk 2]
```diff
@@ -41,18 +41,19 @@ Storybook は視覚的なリグレッションテスト用の素晴らしいツ
...
+`src/components/Task.js` を以下のように変更します:
 
 ```diff:title=src/components/Task.js
```
Comment (chameleonhead) [inline content/intro-to-storybook/react/ja/test.md:46]: この部分のソースコードが英語版と違うようです。見直しをお願いします。

File: content/intro-to-storybook/react/ja/screen.md [hunk 3]
```diff
@@ -323,16 +323,20 @@ Storybook で `error` ストーリーが意図したように動作している
   />
 </video>
 
-## Interactive stories
+## インタラクションテスト
```
Comment (chameleonhead) [inline content/intro-to-storybook/react/ja/screen.md:326]: このページの上部「繋がれた画面」の以下の記述も削除してもらえると良いと思います。

> <div class="aside">
>   💡 <code>src/App.test.js</code> をアップデートすることを忘れないでください。そうしなければ、テストが通らなくなります。
> </div>

File: content/intro-to-storybook/react/ja/screen.md [hunk 4]
```diff
@@ -323,16 +323,20 @@ Storybook で `error` ストーリーが意図したように動作している
...
 
 これまでで、シンプルなコンポーネントから画面まで、完全に機能するアプリケーションを作り上げ、ストーリーを用いてそれぞれの変更を継続的にテストすることができるようになりました。しかし、新しいストーリーを作るたびに、UI が壊れないように、他のすべてのストーリーを手作業でチェックする必要もあります。これは、とても大変な作業です。
```
Comment (chameleonhead) [inline content/intro-to-storybook/react/ja/screen.md:328]: > しかし、新しいストーリーを作るたびに、UI が壊れないように、他のすべてのストーリーを手作業でチェックする必要もあります。

元からだと思いますが、「UIが壊れていなかいどうか」に訳を変更してもらえると嬉しいです。

File: content/intro-to-storybook/react/ja/screen.md [hunk 5]
```diff
@@ -323,16 +323,20 @@ Storybook で `error` ストーリーが意図したように動作している
...
 
 この作業や操作を自動化することはできないのでしょうか？
```
Comment (chameleonhead) [inline content/intro-to-storybook/react/ja/screen.md:330]: > Can't we automate this workflow and test our component interactions automatically?
> この作業や操作を自動化することはできないのでしょうか？

意味はほとんど変わりませんが、ここは訳をあえて省いているのでしょうか。できれば原文に近づけてほしいです。

File: content/intro-to-storybook/react/ja/screen.md [hunk 6]
```diff
@@ -323,16 +323,20 @@ Storybook で `error` ストーリーが意図したように動作している
...
 
+### play 関数を使ったインタラクションテスト
+
 Storybook の [`play`](https://storybook.js.org/docs/react/writing-stories/play-function) 関数が可能にしてくれます。play 関数はストーリーのレンダリング後に実行される小さなコードスニペットを含みます。
```
Comment (chameleonhead) [inline content/intro-to-storybook/react/ja/screen.md:334]: > Storybook's [play](https://storybook.js.org/docs/react/writing-stories/play-function) and [@storybook/addon-interactions](https://storybook.js.org/docs/react/writing-tests/interaction-testing) help us with that. A play function includes small snippets of code that run after the story renders.
> Storybook の [`play`](https://storybook.js.org/docs/react/writing-stories/play-function) 関数が可能にしてくれます。play 関数はストーリーのレンダリング後に実行される小さなコードスニペットを含みます。

ここの訳も微妙に違うので修正をお願いします。

File: content/intro-to-storybook/react/ja/screen.md [hunk 7]
```diff
@@ -323,16 +323,20 @@ Storybook で `error` ストーリーが意図したように動作している
...
 
 play 関数はタスクが更新されたときに UI に何が起こるかを検証するのに役立ちます。フレームワークに依存しない DOM API を使用しています。つまり、 play 関数を使って UI を操作し、人間の行動をシミュレートするストーリーを、フロントエンドのフレームワークに関係なく書くことができるのです。
 
+`@storybook/addon-interactions`は，Storybook のテストを可視化し，段階的なフローを提供するのに役立ちます。さらに，一時停止，再開，巻き戻し，各インタラクションのステップ実行など，便利な UI コントロールが備わっています。
```
Comment (chameleonhead) [inline content/intro-to-storybook/react/ja/screen.md:338]: > step-by-step flow
> 段階的なフロー

ここはなんとなくニュアンスが違う気がします。ここでの step-by-step flow はプログラムの1行ずつの流れを見る的な意味だと思うのでもう少し適切な訳を探してもらえると良いと思います。

File: content/intro-to-storybook/react/ja/screen.md [hunk 8]
```diff
@@ -323,16 +323,20 @@ Storybook で `error` ストーリーが意図したように動作している
...
+
 実際に動かしてみましょう！以下のようにして新しく作成された `InboxScreen` ストーリーを更新し、コンポーネント操作を追加してみましょう:
 
 ```diff:title=src/components/InboxScreen.stories.js
```
Comment (chameleonhead) [inline content/intro-to-storybook/react/ja/screen.md:342]: ここのコードも英語版と異なりますので、英語版に合わせてください。

File: content/intro-to-storybook/react/ja/screen.md [hunk 9]
```diff
@@ -397,7 +401,45 @@ Default.parameters = {
...
 </video>
 
-play 関数を利用して、UI を操作し、タスクを更新した場合の反応を素早く確認することができます。これによって、余計な手間をかけずに UI の一貫性を保つことができます。テスト環境を立ち上げたり、パッケージを追加したりする必要はありません。
+### テスト自動化
+
+play 関数を利用して、UI を操作し、タスクを更新した場合の反応を素早く確認することができます。これによって、余計な手間をかけずに UI の一貫性を保つことができます。
+
+しかし，Storybook をよく見ると，ストーリーを見るときだけインタラクションテストが実行されることがわかります。そのため，変更時に各ストーリーを全てチェックしなければなりません。これは自動化できないのでしょうか？
+
+可能です！Storybook の [test runner](https://storybook.js.org/docs/react/writing-tests/test-runner) は可能にしてくれます。それは [Playwright](https://playwright.dev/) によって実現されたスタンドアロンなパッケージで，全てのインタラクションテストを実行し，壊れたストーリーを検知してくれます。
```
Comment (chameleonhead) [inline content/intro-to-storybook/react/ja/screen.md:410]: ここでの test runner は一般名詞としてカタカナで「テストランナー」のほうが良いと思います。

File: content/intro-to-storybook/react/ja/screen.md [hunk 10]
```diff
@@ -397,7 +401,45 @@ Default.parameters = {
...
+
+それではどのように動くのかみてみましょう！次のコマンドでインストールして走らせます:
+
+```bash
+yarn add --dev @storybook/test-runner
+```
+
+次に， `package.json` `scripts` をアップデートし，新しいテストタスクを追加してください:
```
Comment (chameleonhead) [inline content/intro-to-storybook/react/ja/screen.md:418]: 「`package.json` `scripts`」は「`package.json` の `scripts`」としたほうが良いと思います。

**Review by chameleonhead — CHANGES_REQUESTED** [review]
Nice work! It was easy-to-read translation. I left comments on some sentences, so please fix them.
And could you look over and follow [mozilla japan's editorial guide](https://github.com/mozilla-japan/translation/wiki/Editorial-Guideline). Espesially, you should use 「、」 instead of 「，」 and also add a white space between English word and Japanese word.

Sorry for late response. I'm looking forward to fixing. Thank you.

**Comment by tomoyuki-hiranuma** [comment]
@chameleonhead Thank you for reviewing my translations. I probably updated all pointed out. Please review it.
Thank you very much.

If I have still parts of files to point out, please inform me. I will update quickly.

File: content/intro-to-storybook/react/ja/screen.md [hunk 11]
```diff
@@ -11,9 +11,9 @@ commit: 'e6e6cae'
 
 ## 繋がれた画面
 
-このアプリケーションはとても単純なので、作る画面は些細なものです。リモート API からデータを取得し、(Redux から自分でデータを取得する) `TaskList` をラップして、Redux からの `error` フィールドを追加するだけです。
+このアプリケーションはとても単純なので、作る画面は些細なものです。リモート API からデータを取得し、(Redux から自分でデータを取得する) `TaskList` をラップして、 Redux からの `error` フィールドを追加するだけです。
```
Comment (chameleonhead) [inline content/intro-to-storybook/react/ja/screen.md:14]: 「ラップして、 Redux からの」の「、」と「Redux 」の間にはスペース不要です。
※次の行も同様

File: content/intro-to-storybook/react/ja/screen.md [hunk 12]
```diff
@@ -397,7 +397,45 @@ Default.parameters = {
...
 </video>
 
-play 関数を利用して、UI を操作し、タスクを更新した場合の反応を素早く確認することができます。これによって、余計な手間をかけずに UI の一貫性を保つことができます。テスト環境を立ち上げたり、パッケージを追加したりする必要はありません。
+### テスト自動化
+
+play 関数を利用して、UI を操作し、タスクを更新した場合の反応を素早く確認することができます。これによって、余計な手間をかけずに UI の一貫性を保つことができます。
+
+しかし、Storybook をよく見ると、ストーリーを見るときだけインタラクションテストが実行されることがわかります。そのため、変更時に各ストーリーを全てチェックしなければなりません。これは自動化できないのでしょうか？
+
+可能です！Storybook の [テストランナー](https://storybook.js.org/docs/react/writing-tests/test-runner) は可能にしてくれます。それは [Playwright](https://playwright.dev/) によって実現されたスタンドアロンなパッケージで、全てのインタラクションテストを実行し、壊れたストーリーを検知してくれます。
```
Comment (chameleonhead) [inline content/intro-to-storybook/react/ja/screen.md:406]: 細かいですがテストランナーの前後のスペースは不要です。

File: content/intro-to-storybook/react/ja/screen.md [hunk 13]
```diff
@@ -397,7 +397,45 @@ Default.parameters = {
...
+最後に、Storybook を起動し、新しいターミナルで以下のコマンドを実行してください:
+
+```bash
+yarn test-storybook --watch
+```
+
+<div class="aside">
+💡 play 関数でのインタラクションテストはUIコンポーネントをテストするための素晴らしい手法です。ここで紹介したもの以外にも、さまざまなことができます。もっと深く学ぶには<a href="https://storybook.js.org/docs/react/writing-tests/interaction-testing">公式ドキュメント</a>を読むことをお勧めします。
+<br />
+テストをさらにもっと深く知るためには、 <a href="/ui-testing-handbook">Testing Handbook</a>をチェックしてみてください。これは開発ワークフローを加速させるために、スケーラブルなフロントエンドチームが採用しているテスト戦略について解説しています。
```
Comment (chameleonhead) [inline content/intro-to-storybook/react/ja/screen.md:433]: Testing Handbook の前のスペースは不要、後ろのスペースは必要です。

File: content/intro-to-storybook/react/ja/simple-component.md [hunk 14]
```diff
@@ -260,36 +260,54 @@ export default function Task({ task: { id, title, state }, onArchiveTask, onPinT
 
 見た通り、コンポーネントだけを切り離して作り始めるのは早くて簡単です。あらゆる状態を掘り下げてテストできるので、高品質で、バグが少なく、洗練された UI を作ることができることでしょう。
 
-## 自動化されたテスト
+## アクセシビリティの問題の検知
 
-Storybook はアプリケーションの UI を作成する際に目視でテストする素晴らしい方法を与えてくれます。ストーリーがあれば、タスクの外観が壊れていないことを確認しながらアプリケーションを開発できます。しかしこれは完全に手動の作業なので、警告やエラーがなく表示されていることを、それぞれの状態を確認しながら誰かがクリックしていかなければなりません。なんとか自動化できないものでしょうか。
+アクセシビリティテストとは、[WCAG](https://www.w3.org/WAI/standards-guidelines/wcag/)のルールと他の業界で認めれたベストプラクティスに基づく経験則に対して、自動化ツールを用いることでレンダリングされた DOM を監視することを指します。これは視覚障害、聴覚障害、認知障害などの障害をお持ちの方を含む、できるだけ多くのユーザーがアプリケーションを利用できるように、明らかなアクセシビリティの違反を検知するために QA の第一線として機能します。
```
Comment (chameleonhead) [inline content/intro-to-storybook/react/ja/simple-component.md:265]: WCAG の後ろにスペースを追加してください。

File: content/intro-to-storybook/react/ja/simple-component.md [hunk 15]
```diff
@@ -260,36 +260,54 @@ export default function Task({ task: { id, title, state }, onArchiveTask, onPinT
...
 
-### スナップショットテスト
+Storybook には公式の[アクセシビリティアドオン](https://storybook.js.org/addons/@storybook/addon-a11y)があります。これは、Deque の[axe-core](https://github.com/dequelabs/axe-core)を使っており、[WCAG の問題の 57%](https://www.deque.com/blog/automated-testing-study-identifies-57-percent-of-digital-accessibility-issues/)に対応しています。
```
Comment (chameleonhead) [inline content/intro-to-storybook/react/ja/simple-component.md:267]: axe-core の前後にスペースを入れてください。57% の後ろにスペースを入れてください。

File: content/intro-to-storybook/react/ja/simple-component.md [hunk 16]
```diff
@@ -260,36 +260,54 @@ export default function Task({ task: { id, title, state }, onArchiveTask, onPinT
...
+  features: {
+    interactionsDebugger: true,
+  },
+};
 ```
 
-これだけです。`yarn test` を実行すれば以下のような出力が得られます:
+![Task accessibility issue in Storybook](/intro-to-storybook/finished-task-states-accessibility-issue.png)
+
+ストーリーを一通り見てみると、このアドオンが一つのアクセシビリティの問題を検知したことがわかります。[**「Elements must have sufficient color contrast」**](https://dequeuniversity.com/rules/axe/4.4/color-contrast?application=axeAPI)というエラーメッセージは本質的にタイトルと背景に十分なコントラストがないことを指しています。そのため、アプリケーションの CSS (`src/index.css`にある)の中で、テキストカラーをより暗いグレーに修正する必要があります。
```
Comment (chameleonhead) [inline content/intro-to-storybook/react/ja/simple-component.md:300]: `src/index.css` の後ろにスペースを入れてください。

**Review by chameleonhead — CHANGES_REQUESTED** [review]
@tomoyuki-hiranuma Thank you for updating. Some usage of white space need to be fixed. Please check comments, please.

**Comment by chameleonhead** [comment]
@tomoyuki-hiranuma 
And please update `gatsby-config.js` as follows.

```diff
      tutorialStatus: {
        'intro-to-storybook': {
          react: {
            de: 5.3,
            en: 6.5,
            es: 5.3,
            fr: 6.1,
-           ja: 6.4,
+           ja: 6.5,
            ko: 6.4,
            nl: 5.3,
            pt: 5.3,
            'zh-CN': 6.1,
            'zh-TW': 6.3,
            ar: 6.3,
          },
```

**Comment by tomoyuki-hiranuma** [comment]
@chameleonhead  Thank you for your comments.
I understand the rule of documentation in this PR.
I fixed them, please check them again.

Thank you for your support.

**Review by chameleonhead — APPROVED** [review]

**Comment by chameleonhead** [comment]
@jonniebigodes 
I think it is ok to merge. Please check and merge this PR.

**Pull request merged**